server/src/main/java/org/apache/druid/server/AsyncManagementForwardingServlet.java

```diff
@@ -185,11 +191,11 @@ protected void onServerResponseHeaders(
     super.onServerResponseHeaders(clientRequest, proxyResponse, serverResponse);
   }
 
-  private void handleBadRequest(HttpServletResponse response, String errorMessage) throws IOException
+  private void handleBadRequest(HttpServletResponse response, String errorMessage, int statusCode) throws IOException
```

Comment:
Think this function is for BadRequest type responses specifically, should we add another one for service errors?

Reply:
Yeah, it's only used in two places in this class, so I just renamed it to `handleInvalidRequest`.